bluesky/plan_stubs.py

```diff
@@ -295,6 +302,82 @@ def inner_mvr():
 movr = mvr  # synonym
 
 
+def rd(obj, *, default_value=0):
+    """Reads a single-value non-triggered object
+
+    This is a helper plan to get the scalar value out of a Device (such as an EpicsMotor or a
+    single EpicsSignal).
+
+    For devices that have more than one read key the following rules are used:
+
+    - if exactly 1 field is hinted that value is used
+    - if no fields are hinted and there is exactly 1 value in the
+      reading that value is used
+    - if more than one field is hinted an Exception is raised
+    - if no fields are hinted and there is more than one key in the reading an
+      Exception is raised
+
+    The devices is not triggered and this plan does not create any Events
+
+    Parameters
+    ----------
+    obj : Device
+        The device to be read
+
+    default_value : Any
+        The value to return when not running in a "live" RunEngine (when
+        ret = yield from obj.read() returns None)
+
+    Returns
+    -------
+    val : Any or None
+        The "single" value of the device
+
+    """
+    hints = obj.hints.get("fields", [])
+    if len(hints) > 1:
+        msg = (
+            f"Your object {obj} ({obj.name}.{obj.dotted_name}) "
+            + f"has {len(hints)} items hinted ({hints}).  We "
+            + "do not know how to pick out a single value.  Please "
+            "adjust the hinting by setting the kind of the components of "
+            "this device or by rd ing one of it's components"
+        )
+        raise ValueError(msg)
+    elif len(hints) == 0:
+        if hasattr(obj, "read_attrs"):
+            if len(obj.read_attrs) != 1:
```

Comment:
We have so far avoided referencing `read_attrs` in bluesky. It is not part of the ophyd—bluesky interface; it is an ophyd implementation detail from blueksy’s POV. Would it be alright to do this check by looking at the length of the reading? It is slightly better to fail before we do the reading, but maybe not worth adding this to the interface (even the optional interface).

Reply:
https://github.com/bluesky/bluesky/pull/1313#issuecomment-596443877 <- answered there, resolving this to not split the conversation (are lose it to eventual comment folding).